website/addons/mendeley/model.py

```diff
@@ -68,11 +72,17 @@ def _verify_client_validity(self):
         try:
             self._client.folders.list()
         except MendeleyApiException as error:
-            self._client = None
-            if error.status == 403:
-                raise HTTPError(403)
+            if error.status == 401 and 'Token has expired' in error.message:
+                try:
+                    self.refresh_oauth_key()
```

Comment:
Still needs to ensure that this actually refreshes the key.

From the outdated comment:
>[This call] isn't guaranteed to succeed; either another Mendeley API call needs to be made to see if it worked, or `refresh_oauth_key` needs to be refactored to return a boolean if it doesn't raise anything.